Index: \quasar\src\mixins\panel.js
===================================================================
--- \quasar\src\mixins\panel.js
+++ \quasar\src\mixins\panel.js
@@ -44,9 +44,10 @@
 
     keepAlive: Boolean,
     keepAliveInclude: [ String, Array, RegExp ],
     keepAliveExclude: [ String, Array, RegExp ],
-    keepAliveMax: Number
+    keepAliveMax: Number,
+    allPanels: Boolean
   },
 
   data () {
     return {
@@ -215,9 +216,21 @@
       const panel = this.__isValidPanelName(this.value) &&
         this.__updatePanelIndex() &&
         this.panels[this.panelIndex]
 
-      const content = this.keepAlive === true
+      const content = this.allPanels === true // ensure all panels are always mounted - to avoid form validation rules misses
+        ? this.panels.map((panelItem, panelIndex) =>
+          h('div', {
+            directives: [{name: 'show', value: panelIndex === this.panelIndex}],
+            staticClass: 'q-panel scroll',
+            key: panelItem.key,
+            attrs: { role: 'tabpanel' },
+            // stop propagation of content emitted @input
+            // which would tamper with Panel's model
+            on: cache(this, 'stop', { input: stop })
+          }, [ panelItem ])
+        )
+        : this.keepAlive === true
         ? [
           h('keep-alive', { props: this.keepAliveProps }, [
             h(
               this.needsUniqueWrapper === true
@@ -243,11 +256,12 @@
         ]
 
       return this.animated === true
         ? [
-          h('transition', {
+          h(content.length > 1 ? 'transition-group' : 'transition', {
             props: {
-              name: this.panelTransition
+              name: this.panelTransition,
+              mode: 'out-in'
             }
           }, content)
         ]
         : content
